lib/api/controllers/util/requestAssertions.js

```diff
@@ -80,6 +80,15 @@ module.exports = {
   /**
    * @param {Request} request
    */
+  assertIdStartsNotUnderscore: function requestAssertHasId (request) {
```

Comment:
Well... Technically, an undefined or null ID does not start with an underscore.
So I personally would modify this method to return without throwing if the id is not defined, allowing to get rid of all the `if (request.input.resource.id) {` lines embedding this assertion call.